src/rust/lib/frp/src/io/js.rs

```diff
@@ -0,0 +1,177 @@
+/// Keyboard event listener which calls the callback function as long it lives.
+#[derive(Derivative)]
+#[derivative(Debug(bound=""))]
+pub struct Listener<Callback:?Sized> {
+    logger     : Logger,
+    callback   : Closure<Callback>,
+    element    : web::Window,
+    event_type : String
+}
+
+impl<Callback:?Sized> Listener<Callback> {
+    /// Constructor.
+    pub fn new(logger:impl AnyLogger,event_type:impl Str, callback:Closure<Callback>) -> Self {
+        let element     = web::window();
+        let js_function = callback.as_ref().unchecked_ref();
+        let logger      = Logger::sub(logger,"Listener");
+        let event_type  = event_type.as_ref();
+        let options     = event_listener_options();
+        let result      = element.add_event_listener_with_callback_and_add_event_listener_options
+            (event_type,js_function,&options);
+        if let Err(err) = result {
+            warning!(logger, "Couldn't add {event_type} event listener: {err:?}.");
+        }
+        let event_type = event_type.into();
+        Self {callback,element,event_type,logger}
+    }
+}
+
+impl Listener<dyn KeyboardEventCallback> {
+    /// Creates a new key down event listener.
+    pub fn new_key_down<F>(logger:impl AnyLogger, f:F) -> Self
+    where F : KeyboardEventCallback {
+        let boxed   = Box::new(f);
+        let closure = Closure::<dyn KeyboardEventCallback>::wrap(boxed);
+        Self::new(logger,"keydown",closure)
+    }
+
+    /// Creates a new key up event listener.
+    pub fn new_key_up<F>(logger:impl AnyLogger, f:F) -> Self
+    where F : KeyboardEventCallback {
+        let boxed   = Box::new(f);
+        let closure = Closure::<dyn KeyboardEventCallback>::wrap(boxed);
+        Self::new(logger,"keyup",closure)
+    }
+}
+
+impl Listener<dyn BlurEventCallback> {
+    /// Creates a blur event listener.
+    pub fn new_blur<F>(logger:impl AnyLogger, f:F) -> Self
+    where F : BlurEventCallback {
+        let boxed   = Box::new(f);
+        let closure = Closure::<dyn BlurEventCallback>::wrap(boxed);
+        Self::new(logger,"blur",closure)
+    }
+}
+
+impl<Callback:?Sized> Drop for Listener<Callback> {
+    fn drop(&mut self) {
+        let callback = self.callback.as_ref().unchecked_ref();
+        if self.element.remove_event_listener_with_callback(&self.event_type, callback).is_err() {
+            self.logger.warning("Couldn't remove event listener.");
+        }
+    }
+}
+
+
+// === Event Listener Options ===
+
+/// Retrun options for addEventListener function. See also
+/// https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
+fn event_listener_options() -> web_sys::AddEventListenerOptions {
+    let mut options = web_sys::AddEventListenerOptions::new();
+    // We listen for events in capture phase, so we can decide ourself if it should be passed
+    // further.
+    options.capture(true);
+    // We want to prevent default action on wheel events, thus listener cannot be passive.
```

Comment:
Could you add some links to good explanation what passive events are?

Reply:
There are no passive events, only listeners.
This information is on already linked docs to the addEventListener.